client/src/main/java/org/apache/rocketmq/client/ClientConfig.java

```diff
@@ -241,6 +247,15 @@ public void setLanguage(LanguageCode language) {
     }
 
     public String getNamespace() {
+        if (StringUtils.isNotEmpty(namespace)) {
+            return namespace;
+        }
+
+        if (StringUtils.isNotEmpty(this.namesrvAddr)) {
+            if (NameServerAddressUtils.validateInstanceEndpoint(namesrvAddr)) {
+                return NameServerAddressUtils.parseInstanceIdFromEndpoint(namesrvAddr);
+            }
+        }
         return namespace;
```

Comment:
Actually,I think the last code('return namespace') will return null!I think,so you can return null directly!

Reply:
just keep insistent with the original get method.